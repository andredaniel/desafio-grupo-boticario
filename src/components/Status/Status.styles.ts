import styled, { css } from 'styled-components'
import { theme } from '../../theme'
import { IStatus } from '../../types/status.interface'

export const StatusWrapper = styled.div<IStatus>`
  align-items: center;
  border-radius: 10px;
  display: inline-flex;
  font-size: 0.8rem;
  font-weight: bold;
  padding: 0.5rem 1rem;
  text-transform: uppercase;

  svg {
    height: 1.25em;
    margin-right: 5px;
    width: 1.25em;
  }

  ${({ type }) => {
    switch (type) {
      case 'APPROVED':
        return css`
          background-color: ${theme.colors.success.CONTRAST};
          color: ${theme.colors.success.LIGHT};
        `

      case 'DISAPPROVED':
        return css`
          background-color: ${theme.colors.danger.CONTRAST};
          color: ${theme.colors.danger.LIGHT};
        `

      case 'IN_VALIDATION':
        return css`
          background-color: ${theme.colors.info.CONTRAST}77;
          color: ${theme.colors.info.LIGHT};
        `

      default:
        return css`
          background-color: ${theme.colors.secondary.CONTRAST};
          color: ${theme.colors.secondary.LIGHT};
        `
    }
  }}
`
