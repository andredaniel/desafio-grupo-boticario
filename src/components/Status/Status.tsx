import React from 'react'
import { FiClock, FiThumbsDown, FiThumbsUp } from 'react-icons/fi'
import { IStatus, StatusTypes } from '../../types/status.interface'
import { StatusWrapper } from './Status.styles'

const Status: React.FC<IStatus> = (props): JSX.Element => {
  const { type } = props

  const Icon = (): JSX.Element => {
    switch (type) {
      case StatusTypes.APPROVED:
        return <FiThumbsUp />

      case StatusTypes.DISAPPROVED:
        return <FiThumbsDown />

      case StatusTypes.IN_VALIDATION:
        return <FiClock />

      default:
        return <span />
    }
  }

  return (
    <StatusWrapper {...props}>
      <Icon /> {type}
    </StatusWrapper>
  )
}

export { Status }
