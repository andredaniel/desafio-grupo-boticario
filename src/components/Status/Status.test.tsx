import { render } from '@testing-library/react'
import React from 'react'
import { Status } from './Status'

describe('<Status />', () => {
  it('should render Status component with no props', () => {
    const { container } = render(<Status />)
    expect(container).toMatchSnapshot()
  })

  it('should render Status component with APPROVED status', () => {
    const { container } = render(<Status type="APPROVED" />)
    expect(container).toMatchSnapshot()
  })

  it('should render Status component with DISAPPROVED status', () => {
    const { container } = render(<Status type="DISAPPROVED" />)
    expect(container).toMatchSnapshot()
  })

  it('should render Status component with IN_VALIDATION status', () => {
    const { container } = render(<Status type="IN_VALIDATION" />)
    expect(container).toMatchSnapshot()
  })
})
