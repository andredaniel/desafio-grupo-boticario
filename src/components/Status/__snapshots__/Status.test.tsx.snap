// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should render Status component with APPROVED status 1`] = `
<div>
  <div
    class="sc-bdfBwQ efqQiE"
    type="APPROVED"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
      />
    </svg>
     
    Aprovado
  </div>
</div>
`;

exports[`<Status /> should render Status component with DISAPPROVED status 1`] = `
<div>
  <div
    class="sc-bdfBwQ bvpPMx"
    type="DISAPPROVED"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 15v4a3 3 0 0 0 3 3l4-9V2H5.72a2 2 0 0 0-2 1.7l-1.38 9a2 2 0 0 0 2 2.3zm7-13h2.67A2.31 2.31 0 0 1 22 4v7a2.31 2.31 0 0 1-2.33 2H17"
      />
    </svg>
     
    Reprovado
  </div>
</div>
`;

exports[`<Status /> should render Status component with IN_VALIDATION status 1`] = `
<div>
  <div
    class="sc-bdfBwQ jlsfvO"
    type="IN_VALIDATION"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <polyline
        points="12 6 12 12 16 14"
      />
    </svg>
     
    Em validação
  </div>
</div>
`;

exports[`<Status /> should render Status component with no props 1`] = `
<div>
  <div
    class="sc-bdfBwQ cZGtNA"
  >
    <span />
     
  </div>
</div>
`;
